rosbag2_performance/rosbag2_performance_benchmarking/CMakeLists.txt

```diff
@@ -36,25 +36,25 @@ if(BUILD_ROSBAG2_BENCHMARKS)
     src/result_utils.cpp
     src/results_writer.cpp)
 
-  ament_target_dependencies(writer_benchmark
-    rclcpp
-    std_msgs
-    rosbag2_compression
-    rosbag2_cpp
-    rosbag2_storage
-    yaml_cpp_vendor
+  target_link_libraries(writer_benchmark
+    rclcpp::rclcpp
+    ${std_msgs_TARGETS}
```

Comment:
Actually - this makes me very suspicious of this change! As an end user, how am I supposed to know when I need `${pkg_TARGETS}` vs `pkg::pkg`? This seems strictly worse!

Reply:
So this is a bug in the current CMake code that is generated for message packages (and is one of the main reasons this has been a "soft" changeover so far).

That is, we really, really want this to be `std_msgs::std_msgs`, but we have so far been unable to make that work properly (and we have tried).  For now, the workaround has been to stick with `x_msgs_TARGETS` for all msgs packages, and use modern CMake targets for everything else.

Whether you want to reject this PR because of that, I'll leave to you.  I'll point out that we are accepting this elsewhere in the core for now.